### ISSUE TITLE
State change does not work when state changes by dragging

### ISSUE THREAD

**Issue by wavermartijn** [joachimdalen/azdevops-auto-state#38]
This plugin is what we need. However, I found out it only works for now when you change the state in the form. When you change the state by dragging (the task) on the planboard, the parent state (user story) remains the same. Hope this can be fixed.

Create a user story with some tasks. Drag task to new column. Configured action does not happen. 

**Expected behavior**
The state of the user story (parent) should change to the state as configured

**What version are you on:**
1.2.1

**Comment by joachimdalen** [comment]
Hi there!

I had some issues getting this to work properly during development, so only having it work from the form became the MVP for V1 of this extension. I can take a look again and see if I can get it working in the different sprint and backlog boards.

**Comment by joachimdalen** [comment]
I have done some additional research on this and based on [previous feedback on the forums](https://developercommunity.visualstudio.com/t/Azure-DevOps-work-item-notification-exte/1645588#T-N1646378) there is no contributions in Azure DevOps that allows the extension to be notified on work item changes in these views. 

Neither of my attempts to get it working has been successful either. I submitted a feature request for these contributions, so I'll keep this issue updated as I get more information.

**Comment by wavermartijn** [comment]
Thanks a lot for your speedy follow up. It's unfortunate that the DevOps notifications lack event support for this usecase. Most of the users of the Kanban board will update the task by dragging.

**Comment by joachimdalen** [comment]
I agree, supporting that usecase would be a great addition to this extension.. being limited to only the work item form will make this extension quite limited.

I'll post the link to the feature request here as well in case anyone comes across this and wants to stay updated: https://developercommunity.visualstudio.com/t/Extension:-Hook-into-work-item-changes-o/1676440

**Comment by billysoderlund** [comment]
Service hooks are triggered when data is moved between columns. Maybe it can be used in some way to solve this?

**Comment by joachimdalen** [comment]
Hi @billysoderlund!

Thank you for the suggestion, however, service hooks are intended for external integrations. As far as my knowledge goes, there are no way of executing scripts included in extensions. Service hooks would therefor not be a feasible for this issue.

**Comment by ridhika1628** [comment]
Hi @joachimdalen - Have installed this extension in my organization and used a preset rule i.e. update the PBI state to committed when task gets in progress. However it's not working when am trying to use it in existing PBI. 
Steps I followed -:
1. Add and enable preset Rule in my project
2. Create new child task under PBI which is in New state.
3. Update the state of Child task from 'To Do' to 'In progress'
4. Nothing happened in my parent PBI

**Comment by joachimdalen** [comment]
@ridhika1628 Are you updating the work item from the work item form, and not by dragging it? Did you also do a full refresh of the page after adding the rule

**Comment by ridhika1628** [comment]
@joachimdalen - Yes am updating task in the form not by dragging it. Still my parent i.e. Product backlog item is not getting updated. I did the full refresh as well.  Not sure what's wrong in there. I really want to use your extension as it's the only solution I get.
